pkg/machine/apple/apple.go

```diff
@@ -351,6 +330,68 @@ func StartGenericAppleVM(mc *vmconfigs.MachineConfig, cmdBinary string, bootload
 	return cmd.Process.Release, returnFunc, nil
 }
 
+func getFirstBootAppleVMIgnition(mc *vmconfigs.MachineConfig) ([]string, error) {
+	machineDataDir, err := mc.DataDir()
+	if err != nil {
+		return nil, err
+	}
+
+	// If this is the first boot of the vm, we need to add the vsock
+	// device to vfkit so we can inject the ignition file
+	socketName := fmt.Sprintf("%s-%s", mc.Name, ignitionSocketName)
+	ignitionSocket, err := machineDataDir.AppendToNewVMFile(socketName, &socketName)
+	if err != nil {
+		return nil, err
+	}
+	if err := ignitionSocket.Delete(); err != nil {
+		logrus.Errorf("unable to delete ignition socket: %q", err)
+	}
+
+	ignitionVsockDeviceCLI, err := GetIgnitionVsockDeviceAsCLI(ignitionSocket.GetPath())
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.Debugf("serving ignition file over %s", ignitionSocket.GetPath())
+	go func() {
+		if err := ServeIgnitionOverSock(ignitionSocket, mc); err != nil {
+			logrus.Error(err)
+		}
+		logrus.Debug("ignition vsock server exited")
+	}()
+
+	return ignitionVsockDeviceCLI, nil
+}
+
+func getFirstBootAppleVMCloudInit(mc *vmconfigs.MachineConfig) ([]string, error) {
+	sshKey, err := machine.GetSSHKeys(mc.SSH.IdentityPath)
+	if err != nil {
+		return nil, err
+	}
+
+	tempDir, err := os.MkdirTemp("", "cloud-init-macadam-")
```

Comment:
This is never deleted as far as I can tell?  maybe podman has a directory where this can go, such as `machineDataDir, err := mc.DataDir()`?

Reply:
I followed what they do with the ignition part. I delete the existing user-data if there is one existing. Is that enough? 🤔

Reply:
Should  be good, especially if you put it in datadir.